test/library/draft/Vector/Vector.chpl

```diff
@@ -0,0 +1,1263 @@
+        contained in this vector.
+      :type other: `[?d] eltType`
+    */
+    proc ref extend(other: [?d] eltType) lifetime this < other {
+      on this {
+        _enter();
+        _requestCapacity(_size + d.size);
+        _extendGeneric(other);
+        _leave();
+      }
+    }
+
+    /*
+      Extends this vector by appending a copy of each element yielded by a
+      range.
+
+      .. note::
+
+        Attempting to initialize a vector from an unbounded range will trigger
+        a compiler error.
+
+      :arg other: The range to initialize from.
+      :type other: `range(eltType)`
+    */
+    proc ref extend(other: range(eltType, ?b, ?d)) lifetime this < other {
+      if !isBoundedRange(other) {
+        param e = this.type:string;
+        param f = other.type:string;
+        param msg = "Cannot extend " + e + " with unbounded " + f;
+        compilerError(msg);
+      }
+      on this {
+        _enter();
+        _requestCapacity(_size + other.size);
+        _extendGeneric(other);
+        _leave();
+      }
+    }
+
+    /*
+      Iterate over the elements of this vector.
+
+      :yields: A reference to one of the elements contained in this vector.
+    */
+    iter these() ref {
+      for i in 0..#_size {
+        yield _data[i];
+      }
+    }
+
+    pragma "no doc"
+    iter these(param tag: iterKind) ref where tag == iterKind.standalone {
+      const osz = _size;
+      const minChunkSize = 64;
+      const hasOneChunk = osz <= minChunkSize;
+      const numTasks = if hasOneChunk then 1 else here.maxTaskPar;
+      const chunkSize = floor(osz / numTasks):int;
+      const trailing = osz - chunkSize * numTasks;
+
+      coforall tid in 0..#numTasks {
+        var chunk = _computeChunk(tid, chunkSize, trailing);
+        for i in chunk(0) do
+          yield this[i];
+      }
+    }
+
+    pragma "no doc"
+    proc _computeChunk(tid, chunkSize, trailing) {
+      var lo, hi = 0;
+
+      if tid <= 0 {
+        lo = 0;
+        hi = chunkSize + trailing - 1;
+      } else {
+        lo = chunkSize * tid + trailing;
+        hi = lo + chunkSize - 1;
+      }
+
+      return (lo..hi,);
+    }
+
+    pragma "no doc"
+    iter these(param tag) ref where tag == iterKind.leader {
+      const osz = _size;
+      const minChunkSize = 32;
+      const hasOneChunk = osz <= minChunkSize;
+      const numTasks = if hasOneChunk then 1 else dataParTasksPerLocale;
+      const chunkSize = floor(osz / numTasks):int;
+      const trailing = osz - chunkSize * numTasks;
+
+      // TODO: We can just use the range iterator like above.
+      coforall tid in 0..#numTasks {
+        var chunk = _computeChunk(tid, chunkSize, trailing);
+        yield chunk;
+      }
+    }
+
+    pragma "no doc"
+    iter these(param tag, followThis) ref where tag == iterKind.follower {
+
+      //
+      // TODO: A faster scheme would access the _ddata directly to avoid
+      // the penalty of logarithmic indexing over and over again.
+      //
```

Comment:
Does this comment apply to vector?

Reply:
https://github.com/chapel-lang/chapel/pull/16048/commits/70eb58d56c42f99a7cce68bac906706605a9c41d